framework/core/src/Forum/Controller/LogOutController.php

```diff
@@ -113,4 +123,33 @@ public function handle(Request $request): ResponseInterface
 
         return $this->rememberer->forget($response);
     }
+
+    protected function sanitizeReturnUrl(string $url, string $base): Uri
+    {
+        if (empty($url)) {
+            return new Uri($base);
+        }
+
+        try {
+            $parsedUrl = new Uri($url);
+        } catch (\InvalidArgumentException $e) {
+            return new Uri($base);
+        }
+
+        if (in_array($parsedUrl->getHost(), $this->getWhitelistedRedirectDomains())) {
+            return $parsedUrl;
+        }
+
+        return new Uri($base);
+    }
+
+    protected function getWhitelistedRedirectDomains(): array
```

Comment:
Though not going into the politics of whitelisting. Can we use getAllowedRedirectDomains here?